### PR TITLE
Metric Endpoint Plugin SDK changes

### LINKED ISSUE #611: Enhance ServingSDK to support mertic endpoint and convert prometheus metric collector as plugin
## Is your feature request related to a problem? Please describe.

 - The current metric API implementation creates a dependency on [Prometheus packages](https://github.com/pytorch/serve/blob/master/frontend/gradle.properties#L5)

 - TorchServe should allow users to add different metrics collection frameworks similar to Prometheus.
 
 - Prometheus metric collector should be available as a plugin and not as an inbuilt framework.

## Describe the solution

 - Enhance ServingSDK to support mertic endpoint

 - Convert prometheus metric collector as plugin

### PR DESCRIPTION
**Pull request by maheshambule**
## Description

This PR adds contracts/interfaces in the SDK for Metric Endpoint Plugin support.

Fixes #611 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing
- UT/IT execution results

- Logs

## Checklist:

- [ X] Have you added tests that prove your fix is effective or that this feature works?
- [ X] New and existing unit tests pass locally with these changes?
- [ X] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: serving-sdk/src/main/java/org/pytorch/serve/servingsdk/metrics/MetricLogEvent.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+     */
+    BaseMetric getMetric();
+
+    /**
+     * Get the log Timestamp
+     * @return The Timestamp
+     */
+    Date getTimestamp();
+}
+
```
Comment (maaquib) [inline serving-sdk/src/main/java/org/pytorch/serve/servingsdk/metrics/MetricLogEvent.java:27]: NIT: Unnecessary newlines

File: serving-sdk/src/main/java/org/pytorch/serve/servingsdk/metrics/DimensionRegistry.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package org.pytorch.serve.servingsdk.metrics;
+
+/**
+ * This is a registry for different metric dimensions and their values.
+ * This is not exhaustive list. Dimension can be added to Metric without adding it to this Registry.
+ */
+public class DimensionRegistry {
```
Comment (maaquib) [inline serving-sdk/src/main/java/org/pytorch/serve/servingsdk/metrics/DimensionRegistry.java:7]: NIT: Not quite clear where `DimensionRegistry` and `InbuiltMetricsRegistry` would be used

**Review by maaquib — COMMENTED** [review]
LGTM for most part. A couple of comments

File: serving-sdk/src/main/java/org/pytorch/serve/servingsdk/metrics/MetricLogEvent.java (hunk 1 above)
Comment (maheshambule) [reply serving-sdk/src/main/java/org/pytorch/serve/servingsdk/metrics/MetricLogEvent.java]: removed.

File: serving-sdk/src/main/java/org/pytorch/serve/servingsdk/metrics/DimensionRegistry.java (hunk 2 above)
Comment (maheshambule) [reply serving-sdk/src/main/java/org/pytorch/serve/servingsdk/metrics/DimensionRegistry.java:10]: The purpose of these registries is to list and refer all the Metrics/Dimensions available in TorchServe in one place. It is expected that whenever a new metric/dimension gets added in TorchServe, a corresponding entry should be made here. The Plugin writers then should be able to refer to this list, and will not have to worry about the spelling of metric, duplicate metric name, etc. However, currently, it is not mandatory to make an entry here.  I have updated the Doc String to reflect this comment.

**Review by maaquib — APPROVED** [review]

**Comment by maheshambule** [comment]
@dhanainme, could you please have a look at the PR?

**Review by dhanainme — APPROVED** [review]

**Pull request merged**